elasticsearch/includes/classes/class-elasticsearch.php

```diff
@@ -116,6 +124,20 @@ public function action__plugins_loaded() {
 		}
 	}
 
+	public function action__edited_term( $term_id, $tt_id, $taxonomy ) {
+		// Find ID of all attached posts (query lifted from wp_delete_term())
+		$object_ids = (array) $wpdb->get_col( $wpdb->prepare( "SELECT object_id FROM $wpdb->term_relationships WHERE term_taxonomy_id = %d", $tt_id ) );
+
+		// Bulk index them all
+		\ElasticPress\Indexables::factory()->get( 'post' )->bulk_index( $object_ids );
+	}
+
+	public function action__set_object_terms( $object_id, $terms, $tt_ids, $taxonomy, $append, $old_tt_ids ) {
```

Comment:
Would this lead to double indexing in the case of a post update?

Reply:
Good point, yes it would. I've reworked this to use EP's SyncManager queue, which de-dupes on the same request (so can only queue the same post once).